### DISCUSSION TITLE
Support partial TMDL declaration when serializing

### DISCUSSION THREAD

**Discussion by Crumar** [TabularEditor/TabularEditor3#1295] [Ideas]
### Is your feature request related to a problem? Please describe

We want to use TMDL partial declaration, the simplest is example is probably the date table.
We could have a base Date.tmdl and multiple extensions, e.g. one consisting calendar year definitions and one with financial year definitions. Tabular Editor supports loading this, but when serializing the model, the files are merged together into a single one.

### Describe the solution you'd like

When saving model to folder:
- Deletions should be removed from the original file where it was defined
- Modifications should be modified in the original file where it was defined
- Unsure how to handle additions within GUI, maybe source file attribute, custom dialog, ...?

### Goals

1. Partial Declarations / Separations are not lost when saving

### Additional context

I understand this may probably a request to Microsoft, but in the end Tabular Editor has to support it as well to store the origin of each node in the TOM.

**Comment by otykier** [comment]
Related discussion: https://github.com/TabularEditor/TabularEditor3/discussions/1198
